src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb

```diff
@@ -907,11 +918,65 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return result
         End Function
 
+        Private Sub GetLambdaId(syntax As SyntaxNode, closureKind As ClosureKind, closureOrdinal As Integer, ByRef topLevelMethodId As MethodDebugId, ByRef lambdaOrdinal As Integer)
+            Debug.Assert(syntax IsNot Nothing)
+
+            Dim lambdaOrLambdaBodySyntax As SyntaxNode
+            Dim isLambdaBody As Boolean ' indicates whether a node needs mapping for lambda or its bodies
+            Dim syntaxNode = TryCast(syntax, LambdaExpressionSyntax)
+            If syntaxNode IsNot Nothing Then
+                ' lambda
+                isLambdaBody = True
+                lambdaOrLambdaBodySyntax = syntaxNode.SubOrFunctionHeader
+            ElseIf syntax.AncestorsAndSelf().OfType(Of QueryExpressionSyntax)().Any() Then
```

Comment:
In reality, this syntax refers to syntax of BoundLabmda, so it cannot go that far. However, I will update this check as debug.assert, so we don't have to check in the product.